### ISSUE TITLE
dr_flac: handful of minor warnings from clang's static code analysis

### ISSUE THREAD

**Issue by krcroft** [mackron/dr_libs#64]
https://github.com/dreamer/dosbox-staging/commit/974f2abcd6df3eb5cb2200487b288e4aee745b54/checks#step:5:866

https://github.com/dreamer/dosbox-staging/commit/974f2abcd6df3eb5cb2200487b288e4aee745b54/checks#step:6:523

Possibly some false-positives here, but wanted to pass them on.

**Comment by mackron** [comment]
Thanks. Some of these I'll clean up, but there's others that I'd rather leave as-is for code clarity and for safety in future maintenance work.

**Comment by krcroft** [comment]
Understood and makes perfect sense.  
Glad at least some of them are meaningful.

**Comment by mackron** [comment]
Were these warnings from the most recent version of dr_flac? If not, could you do me a favor and send through and updated listing just so I can see what's still relevant?

**Comment by mackron** [comment]
```
dr_flac.h:1275:22: warning: ‘drflac__gIsLZCNTSupported’ defined but not used [-Wunused-variable]
 static drflac_bool32 drflac__gIsLZCNTSupported = DRFLAC_FALSE;
                      ^~~~~~~~~~~~~~~~~~~~~~~~~
```
I think this one has already been fixed in the latest version of dr_flac. I can't see how that would be having warnings, and I vaguely remember seeing this warning during my internal testing and fixing it.

All the other warnings except this one should be cleaned up now. Removing that line has dangerous implications for future maintenance work that may need to use that `bytesSeeked` variable. Do you know if `(void)bytesSeeked` would clean up that warning? If so, I'd be happy to do that.
```
analyze-build: INFO: ./dr_flac.h:6731:13: warning: Value stored to 'bytesSeeked' is never read
analyze-build: INFO:             bytesSeeked += bytesRemainingToSeek;
analyze-build: INFO:             ^              ~~~~~~~~~~~~~~~~~~~~
```
These changes are in the flac/dev branch. Let me know how that looks!

**Comment by krcroft** [comment]
Excellent Dave! 

The statement "(void)bytesSeeked;", after the flagged line does the trick to keep the static analyzer happy while being valid code across all the compilers tested.

Interestingly, gcc 9.1 complains about the following being unused, perhaps due to the #defines in other parts knocking out its use by the preprocessor, before the compiler gets the code.
```
2019-09-27T23:50:49.6287267Z dr_flac.h:1275:22: warning: "drflac__gIsLZCNTSupported" defined but not used [-Wunused-variable]
2019-09-27T23:50:49.6287578Z  1275 | static drflac_bool32 drflac__gIsLZCNTSupported = DRFLAC_FALSE;
2019-09-27T23:50:49.6287729Z       |                      ^~~~~~~~~~~~~~~~~~~~~~~~~
```
One to go.. very tight work considering dr_mp3 and dr_flac are being tested on gcc 9.1, 8.3, 7.4, and 5.4; and CLang 10.0.1 and 6.01, in addition to Clang's static analyzer.

**Comment by mackron** [comment]
I've pushed an update to the flac/dev branch for you to try out.

**Comment by krcroft** [comment]
The static-analysis warning is gone.

gcc 9.1 is now warning free when compiling with:

```
define DR_FLAC_IMPLEMENTATION
#define DR_FLAC_NO_STDIO 1
#define DR_FLAC_NO_WIN32_IO 1
#define DR_FLAC_NO_CRC 1
#define DRFLAC_MALLOC(sz) SDL_malloc((sz))
#define DRFLAC_REALLOC(p, sz) SDL_realloc((p), (sz))
#define DRFLAC_FREE(p) SDL_free((p))
#define DRFLAC_COPY_MEMORY(dst, src, sz) SDL_memcpy((dst), (src), (sz))
#define DRFLAC_ZERO_MEMORY(p, sz) SDL_memset((p), 0, (sz))
#include "dr_flac.h"
```
Output: https://pipelines.actions.githubusercontent.com/RDKYnb9yFkXPfcXINdFWH3mzDaNqlgzrXM4kV8rnx6PJBbLLEF/_apis/pipelines/1/runs/129/signedlogcontent/48?urlExpires=2019-09-28T12%3A23%3A20.5836460Z&urlSigningMethod=HMACV1&urlSignature=iyQs9cS9Zry7GTt4G%2FWcTjqNSUVklCbzAvN4K7HRr40%3D


But when I accidentally enabled STDIO and left SIMD disabled, the warning persisted:
define DR_FLAC_IMPLEMENTATION
#define DR_FLAC_NO_SIMD 1
// #define DR_FLAC_NO_STDIO 1
#define DR_FLAC_NO_WIN32_IO 1
#define DR_FLAC_NO_CRC 1
#define DRFLAC_MALLOC(sz) SDL_malloc((sz))
#define DRFLAC_REALLOC(p, sz) SDL_realloc((p), (sz))
#define DRFLAC_FREE(p) SDL_free((p))
#define DRFLAC_COPY_MEMORY(dst, src, sz) SDL_memcpy((dst), (src), (sz))
#define DRFLAC_ZERO_MEMORY(p, sz) SDL_memset((p), 0, (sz))
#include "dr_flac.h"
```
produced:
```
2019-09-28T08:24:42.9129873Z In file included from flac.c:48:0:
2019-09-28T08:24:42.9131959Z dr_flac.h:1276:22: warning: â€˜drflac__gIsLZCNTSupportedâ€™ defined but not used [-Wunused-variable]
2019-09-28T08:24:42.9132448Z  static drflac_bool32 drflac__gIsLZCNTSupported = DRFLAC_FALSE;
2019-09-28T08:24:42.9132955Z                       ^~~~~~~~~~~~~~~~~~~~~~~~~
```
Output: https://pipelines.actions.githubusercontent.com/RDKYnb9yFkXPfcXINdFWH3mzDaNqlgzrXM4kV8rnx6PJBbLLEF/_apis/pipelines/1/runs/127/signedlogcontent/12?urlExpires=2019-09-28T12%3A07%3A28.9661679Z&urlSigningMethod=HMACV1&urlSignature=roaDitD9XFVvFQGlXU2Q7Db0tBy8VxVKJkSmsqbKAlI%3D

**Comment by mackron** [comment]
Thanks for the info! I didn't realise that variable was tied to the `DR_FLAC_NO_SIMD` option. I wonder what I did that... In any case, I was able to reproduce the warning and fixed it. I've released all of these warning fixes to the master branch now and bumped the version. Thanks for the report!

**Comment by krcroft** [comment]
Subtle indeed. I am upgrading to the released 0.12.1; thank you Dave!
